src/ruby/puppetserver-lib/puppet/server/logging.rb

```diff
@@ -0,0 +1,26 @@
+require 'puppet/server/log_collector'
+
+module Puppet
+  module Server
+    module Logging
+
+      def self.capture_logs(level, &block)
+        old_log_level = Puppet.settings[:log_level]
+        Puppet.settings[:log_level] = level if level
+
+        logs = []
+        result = nil
+        log_dest = Puppet::Server::LogCollector.new(logs)
+        Puppet::Util::Log.with_destination(log_dest) do
+          result = yield
+        end
+
+        log_entries = logs.map(&:to_data_hash)
```

Comment:
With the increase in logging at debug level are we concerned about this additional mapping over the logs, or is that concern yagni at this point?

Reply:
Not sure if we really have a choice? The clojure needs to get a hash back to properly parse to JSON for the response. What were you thinking instead?

Reply:
Updating the LogCollector's `<<` method to call `to_data_hash` on its input so we don't have to map over all the log events after they were collected.

Reply:
Oh yeah that's much better. Thanks.